snapshot/storage/bolt.go

```diff
@@ -372,7 +298,65 @@ func (ms *boltMetastore) Commit(ctx context.Context, key, name string) (id strin
 	return
 }
 
-func getSnapshot(bkt *bolt.Bucket, key string, ss *Snapshot) error {
+func withBucket(ctx context.Context, fn func(context.Context, *bolt.Bucket, *bolt.Bucket) error) error {
+	t, ok := ctx.Value(transactionKey{}).(*boltFileTransactor)
+	if !ok {
+		return errors.Errorf("no transaction in context")
```

Comment:
It might be worth creating constants for these errors (especially this "no transaction in context" that is repeated) or typed errors so that calling code can distinguish between something that is programmer error (not passing a transaction in the context) or something wrong elsewhere (the underlying storage being non-writable).